### ISSUE TITLE
[Curp] Optimize the sync channel blocking mechanism

### ISSUE THREAD

**Issue by markcty** [datenlord/Xline#89]
Currently, the sync of commands that can be executed speculatively will be postponed until the execution is finished. This is to preserve the execution order. For example, if a=1 and a=2 arrive sequentially. Only after a=1 finishes speculative execution will a=1 be synced. Only then will a=2 be synced and executed due to the key-based channel.

The postponement is unnecessary. We can add another mechanism to preserve the execution order. For example, store the execution order in a channel and spawn an execution task.

_Originally posted by @rogercloud in https://github.com/datenlord/Xline/pull/86#discussion_r1033146509_
